Automattic/wp-calypso
# Copy wooexpress referer parameters from woocommerce.com to cookie on wooexpress trial landing page

Related to #

## Proposed Changes
This is from pau2Xa-54t-p2#comment-13966
When a user is on https://woocommerce.com/start/?nuxentrysource=hero_cta#/installation and they click the **"Try WooExpress for Free CTA"**  It leads them to a wordpress.com landing page.

This PR enables wordpress.com to pick up aff and vid parameters from the URL http://wordpress.com/setup/wooexpress/   where the user is redirected after clicking the CTA from above.

## WIP Testing Instructions
- Go to the live testing site [live](https://calypso.live/?image=registry.a8c.com/calypso/app:build-80579)
- Ensure you are logged out.
- Go to /setup/wooexpress/?vid=67386441&aff=86844 on the live testing site
- Verify that the 'aff'  and 'vid' parameters are carried on to the next step in the wooexpress setup flow.

**Test the flow for logged in Users**
- Log on to the live testing site
- Go to /setup/wooexpress/?vid=67386441&aff=86844
- Ensure that registration of a new wooexpress account happens without any issues.

**Test the Flow in a different language**
- On your local calypso install
- Switch to branch add/wooexpress-referer-cookie-tracking
- Change locale on line 53 here to 'fr'. https://github.com/Automattic/wp-calypso/blob/add/wooexpress-referer-cookie-tracking/client/landing/stepper/declarative-flow/trial-wooexpress-flow.ts#L53
- On your local Calypso install, make sure you are logged out.
- Go to http://calypso.localhost:3000/setup/wooexpress/?vid=67386441&aff=86844   and complete the process.
- Ensure that the flow and registration of a new account happen without issues.



*

## Pre-merge Checklist
